oauth2/oauth2-jose/src/main/java/org/springframework/security/oauth2/jwt/JwtIssuerValidator.java

```diff
@@ -28,15 +31,19 @@
  */
 public final class JwtIssuerValidator implements OAuth2TokenValidator<Jwt> {
 
-	private final JwtClaimValidator<String> validator;
+	private final JwtClaimValidator<Object> validator;
 
 	/**
 	 * Constructs a {@link JwtIssuerValidator} using the provided parameters
 	 * @param issuer - The issuer that each {@link Jwt} should have.
 	 */
 	public JwtIssuerValidator(String issuer) {
 		Assert.notNull(issuer, "issuer cannot be null");
-		this.validator = new JwtClaimValidator(JwtClaimNames.ISS, issuer::equals);
+
+		Predicate<Object> testClaimValueURL = (
```

Comment:
Have you already considered

```java
Predicate<Object> testClaimValue = (claimValue) -> 
        claimValue != null && issuer.equals(claimValue.toString())
```

as it simplifies the logic?

Reply:
Hi Josh, many thanks for your feedback. I simplified the logic as you suggested and pushed a fix-up commit. After the checks succeed I'll squash the commits and change the commit comment having "Closes ...".